cmd/clusterctl/client/tree/discovery.go

```diff
@@ -141,25 +143,25 @@ func Discovery(ctx context.Context, c client.Client, namespace, name string, opt
 		if err != nil {
 			return nil, err
 		}
-
-		// Handles orphan machines.
-		if len(machineMap) < len(machinesList.Items) {
-			other := VirtualObject(cluster.Namespace, "OtherGroup", "Other")
-			tree.Add(workers, other)
-
-			for i := range machinesList.Items {
-				m := &machinesList.Items[i]
-				if _, ok := machineMap[m.Name]; ok {
-					continue
-				}
-				addMachineFunc(other, m)
-			}
-		}
 	}
 
 	if len(machinePoolList.Items) > 0 { // Add MachinePool objects
 		tree.Add(cluster, workers)
-		addMachinePoolsToObjectTree(ctx, c, cluster.Namespace, workers, machinePoolList, tree)
+		addMachinePoolsToObjectTree(ctx, c, cluster.Namespace, workers, machinePoolList, machinesList, tree, addMachineFunc)
```

Comment:
this is probably just me not understanding the code but aren't the machines already getting added above? Or is that just for MachineDeployments?

Reply:
Yes, so the code here first finds all the Machines associated with the Cluster and then adds the Machines for a MachineDeployment and for the control plane. Then, we're adding the Machines for a MachinePool, and lastly we add any Machines that don't belong to any category.